bin/await

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+main() {
+
+	local subcommand="${1:-}"
+	shift
+
+	case "$subcommand" in
+		http)
+			local host timeout=60 counter=0 retry=3
+
+			while getopts ":h:t:r:" opt; do
+
+				case "${opt}" in
+					h)
+						host="${OPTARG}"
+						;;
+					t)
+						timeout="${OPTARG}"
+						;;
+					r)
+						retry="${OPTARG}"
+						;;
+					\?)
+						echo "Invalid option: -${OPTARG}" 1>&2
+						return 1
+						;;
+					:)
+						echo "Option -${OPTARG} requires an argument." 1>&2
+						return 1
+						;;
+				esac
+			done
+
+			shift $((OPTIND-1))
+
+			while [ "${counter}" -lt "${retry}" ]; do
+				if curl -s --fail -m "${timeout}" "${host}" > /dev/null; then
```

Comment:
We should include `$@` here as well. This will allow people to set headers and whatever other stuff they like.